driver/ppm_fillers.c

```diff
@@ -4806,3 +4806,106 @@ int f_sys_mkdirat_x(struct event_filler_arguments *args)
 
 	return add_sentinel(args);
 }
+
+int f_sys_fchmodat_x(struct event_filler_arguments *args)
+{
+	unsigned long val;
+	int res;
+	int64_t retval;
+
+	retval = (int64_t)syscall_get_return_value(current, args->regs);
+	res = val_to_ring(args, retval, 0, false, 0);
+	if (unlikely(res != PPM_SUCCESS))
+		return res;
+
+	/*
+	 * dirfd
+	 */
+	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
```

Comment:
`syscall_get_arguments`? the interface is slightly different but at least we avoid adding new deprecated calls.

I see we don't have a wrapper for the new API that works for <5.1 so you'd have to write one, maybe something like:

```
#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 1, 0))
#define syscall_get_all_arguments(task, reg, args) syscall_get_arguments(task, reg, 0, 6, args)
#define syscall_get_arguments_deprecated syscall_get_arguments
#else 
#define syscall_get_all_arguments syscall_get_arguments
#define syscall_get_arguments_deprecated(_task, _reg, _start, _len, _args) \
  // ...
#endif

Reply:
Thanks for the suggestion, will look into this

Reply:
+1 on using the non deprecated call, also it doesn't look like modifying the header will too much work

Reply:
I think that this can be done in another PR after merging this. All the syscalls are using the deprecated one in fillers and if I make the macro for only one filler it would mean I'd have to update all of them. It looks like a completely different task to me